src/transformers/modeling_encoder_decoder.py

```diff
@@ -201,36 +213,70 @@ def save_pretrained(self, save_directory):
             os.mkdir(os.path.join(save_directory, "decoder"))
         self.decoder.save_pretrained(os.path.join(save_directory, "decoder"))
 
-    def forward(self, encoder_input_ids, decoder_input_ids, **kwargs):
-        """ The forward pass on a seq2eq depends what we are performing:
-
-        - During training we perform one forward pass through both the encoder
-          and decoder;
-        - During prediction, we perform one forward pass through the encoder,
-          and then perform several forward passes with the encoder's hidden
-          state through the decoder to decode a full sequence.
-
-        Therefore, we skip the forward pass on the encoder if an argument named
-        `encoder_hidden_state` is passed to this function.
+    def forward(
+        self,
+        input_ids=None,
```

Comment:
The API is very similar to what is done in `T5` and `Bart`. I also thought about just having a single `**kwargs` function argument that is then split into `decoder_kwargs` and `encoder_kwargs`. These kwargs can then just be inserted into their respective models. This would make the function a bit more flexible (mabye people want to use special Bert arguments for `next_sentence_prediction`, etc.), but less readable. A compromise, would maybe be to leave it as it is now and additionally add the **kwargs argument that is split in put into the correct models. What do you think @sshleifer @thomwolf @LysandreJik @yjernite ?